docs/use_bls_provider.md

```diff
@@ -0,0 +1,226 @@
+# BLS Provider
+
+This document will show you how to interact with the `BlsProvider` and `BlsSigner` classes.
+
+The `BlsProvider` and `BlsSigner` are part of the `bls-wallet-clients` npm package, and help you interact with BLS Wallet components in a similar way you would use an Ethers provider and signer to interact with the Ethereum ecosystem. It offers developers a familiar develoment experience, while providing access to BLS Wallet components and features. Essentially it's a Ethers provider-shaped wrapper around `bls-wallet-clients`.
+
+The `BlsProvider` and `BlsSigner` mimic the behaviour of an Ethers [JsonRpcProvider](https://docs.ethers.org/v5/api/providers/jsonrpc-provider/) and [JsonRpcSigner](https://docs.ethers.org/v5/api/providers/jsonrpc-provider/#JsonRpcSigner) respectively. In this implementation, note that the `BlsSigner` has knowledge of its own private key. For more information on Ethers providers and signers, visit the [Ethers v5 docs](https://docs.ethers.org/v5/).
+
+The `BlsProvider` and `BlsSigner` are covered by over 100 test cases, including integration tests. If any functionality is not documented here, it will likely be documented by test cases.
+
+# Creating instances
+
+## BlsProvider
+
+### Instantiating a BlsProvider
+
+```ts
+import { Experimental, BlsWalletWrapper } from "bls-wallet-clients";
+
+const aggregatorUrl = "http://localhost:3000";
+const verificationGateway = "0x123";
+const aggregatorUtilities = "0x321";
+const rpcUrl = "http://localhost:8545";
+const network = {
+    name: "localhost",
+    chainId: 0x539, // 1337
+};
+
+const provider = new Experimental.BlsProvider(
+    aggregatorUrl,
+    verificationGateway,
+    aggregatorUtilities,
+    rpcUrl,
+    network
+);
+```
+
+### BlsSigner
+
+**Important:** Ensure that the BLS wallet you are linking the `BlsSigner` to via the private key is funded. Alternatively, if a wallet doesn't yet exist, it will be lazily created on the first transaction. In this scenario, you can create a random BLS private key with the following helper method and fund that account. It will need to be funded in order to send its first transaction.
+
+```ts
+import { BlsWalletWrapper } from "bls-wallet-clients";
+
+const privateKey = await BlsWalletWrapper.getRandomBlsPrivateKey();
```

Comment:
We should expose this as a static method on `BlsProvider/Signer` so in some use cases users can avoid the wrapper import. Future follow up, out of scope.

Reply:
Makes sense, will create a follow up issue

Reply:
https://github.com/web3well/bls-wallet/issues/565